cli/isolate_state.rs

```diff
@@ -42,9 +49,16 @@ pub struct IsolateState {
   pub modules: Mutex<Modules>,
   pub worker_channels: Option<Mutex<WorkerChannels>>,
   pub global_timer: Mutex<GlobalTimer>,
+  pub workers: Mutex<WebWorkerTable>,
 }
 
+// TODO(afinch7) abstract out IsolateState to a trait, since
+// some isolates don't require or shouldn't be given some
+// state data
```

Comment:
Maybe. Generally I would like to err on the side of concrete types rather than traits because it tends to complicate the code and force generics everywhere. What are you thinking?

Reply:
This is old. I decided against this since most things like `worker_channels` are optional anyways. I don't know if we want to provide user land workers with flags/args in their code, but that could be achieved via other methods that are far simpler.